src/move.test.c

```diff
@@ -295,6 +295,123 @@ static void test_age_trails(CuTest *tc) {
+    fix->u = test_create_unit(fix->f = test_create_faction(0), fix->r=findregion(-1,0));
+    assert(fix->r && fix->r->terrain->flags & SAIL_INTO);
+    set_level(fix->u, SK_SAILING, fix->st_boat->sumskill);
+    u_set_ship(fix->u, fix->sh = test_create_ship(fix->u->region, fix->st_boat));
+    assert(fix->f && fix->u && fix->sh);
+    fix->f->locale = get_or_create_locale("de");
+
+}
+
+static void test_ship_no_overload(CuTest *tc) {
+    struct drift_fixture fix;
+    setup_drift(&fix);
+
+    fix.u->number = 2;
+    movement();
+    CuAssertPtrEquals(tc, fix.u->region, findregion(-1,0));
+    CuAssertIntEquals(tc, 0, fix.sh->damage);
+}
+
+static void test_ship_normal_overload(CuTest *tc) {
+    struct drift_fixture fix;
+    setup_drift(&fix);
+
+    fix.u->number = 21;
+    movement();
+    CuAssertPtrEquals(tc, fix.u->region, findregion(0, 0));
+    CuAssertIntEquals(tc, 2, ship_damage_percent(fix.sh));
+    CuAssertPtrNotNull(tc, test_find_messagetype(fix.f->msgs, "ship_drift"));
+}
+
+static void test_ship_big_overload(CuTest *tc) {
+    struct drift_fixture fix;
+    setup_drift(&fix);
+
+    fix.u->number = 22;
+    movement();
+    CuAssertPtrEquals(tc, fix.u->region, findregion(-1, 0));
+    CuAssertIntEquals(tc, 5, ship_damage_percent(fix.sh));
+    CuAssertPtrNotNull(tc, test_find_messagetype(fix.f->msgs, "massive_overload"));
+}
+
+static void test_ship_no_real_overload(CuTest *tc) {
+    struct drift_fixture fix;
+    setup_drift(&fix);
+
+    fix.u->number = 21;
+    damage_ship(fix.sh, .80);
+    movement();
+    CuAssertPtrEquals(tc, fix.u->region, findregion(0, 0));
+    CuAssertIntEquals(tc, 82, ship_damage_percent(fix.sh));
+    CuAssertPtrEquals(tc, 0, test_find_messagetype(fix.f->msgs, "massive_overload"));
+}
+
+static void test_ship_ridiculous_overload(CuTest *tc) {
+    struct drift_fixture fix;
+    setup_drift(&fix);
+
+    fix.u->number = 500;
+    movement();
+    CuAssertIntEquals(tc, 37, ship_damage_percent(fix.sh));
+    CuAssertPtrNotNull(tc, test_find_messagetype(fix.f->msgs, "massive_overload"));
+}
+
+static void test_ship_ridiculous_overload_no_captain(CuTest *tc) {
+    struct drift_fixture fix;
+    setup_drift(&fix);
+    set_level(fix.u, SK_SAILING, 0);
+
+    fix.u->number = 500;
+    movement();
+    CuAssertIntEquals(tc, 37, ship_damage_percent(fix.sh));
+    CuAssertPtrNotNull(tc, test_find_messagetype(fix.f->msgs, "massive_overload"));
+}
+
+static void test_ship_ridiculous_overload_bad(CuTest *tc) {
+    struct drift_fixture fix;
+    setup_drift(&fix);
+
+    fix.u->number = 500;
+    set_param(&global.parameters, "rules.ship.overload.damage.max", "1");
+    movement();
+    CuAssertTrue(tc, ship_damage_percent(fix.sh) > 99);
+    CuAssertPtrNotNull(tc, test_find_messagetype(fix.f->msgs, "massive_overload"));
+    CuAssertPtrEquals(tc, 0, fix.sh->region);
+    CuAssertPtrNotNull(tc, test_find_messagetype(fix.f->msgs, "shipsink"));
+}
+
+extern double damage_overload(double overload);
```

Comment:
Ist es eine gute Idee diese Funktion aus move.c so zu deklarieren, anstatt in move.h, wenn ich sie nur testen will und nicht "public" brauche?